proto/storage/resource_collection.proto

```diff
@@ -0,0 +1,54 @@
+syntax = "proto3";
+
+option go_package = "storage";
+option java_package = "io.stackrox.proto.storage";
+
+import "google/protobuf/timestamp.proto";
+import "gogoproto/gogo.proto";
+import "storage/user.proto";
+import "storage/policy.proto";
+
+package storage;
+
+message ResourceCollection {
+  message EmbeddedResourceCollection {
+    // this is searchable to force a separate table
+    string id = 1 [(gogoproto.moretags) = 'search:"Embedded Collection ID",hidden'];
+  }
+
+  string id                                                = 1 [(gogoproto.moretags) = 'sql:"pk"']; // primary key
+  string name                                              = 2 [(gogoproto.moretags) = 'search:"Collection Name"'];
+  string description                                       = 3;
+  google.protobuf.Timestamp created_at                     = 4;
+  google.protobuf.Timestamp last_updated                   = 5;
+  SlimUser created_by                                      = 6;
+  SlimUser updated_by                                      = 7;
+  // `resource_selectors` resolve as disjunction (OR) with each-other and with selectors from `embedded_collections`. For MVP, the size of resource_selectors will at most be 1 from UX standpoint.
+  repeated ResourceSelector resource_selectors             = 8;
+  repeated EmbeddedResourceCollection embedded_collections = 9 [(gogoproto.moretags) = 'sql:"fk(EmbeddedResourceCollection:id)"'];
+}
+
+message ResourceSelector {
+  // `rules` resolve as a conjunction (AND).
+  repeated SelectorRule rules = 1;
+}
+
+message SelectorRule {
+  // `field_name` can be one of the following:
+  // - Cluster
+  // - Cluster Label
+  // - Namespace
+  // - Namespace Label
+  // - Namespace Annotation
+  // - Deployment
+  // - Deployment Label
+  // - Deployment Annotation
+  string field_name         = 1;
+  BooleanOperator operator  = 2;
```

Comment:
Add a comment on this that only `OR` is supported for now.

Reply:
Do you mean for this comment to be on the `operator` field? That's where I added it. The field below also has some kind of comment about `AND` and `OR` that came from the design doc that probably should get cleaned up